File: ios/App/Sources/Core/Domain/NavCaddy/MissBias.swift
import Foundation

/// Represents the miss bias tendency for a club.
///
/// Spec reference: player-profile-bag-management.md R4
/// Plan reference: player-profile-bag-management-plan.md Task 2
struct MissBias: Codable, Hashable {
    let dominantDirection: MissBiasDirection
    let missType: MissType?
    let isUserDefined: Bool
    /// Confidence level (0.0-1.0).
    let confidence: Double
    let lastUpdated: Date

    init(
        dominantDirection: MissBiasDirection,
        missType: MissType? = nil,
        isUserDefined: Bool,
        confidence: Double,
        lastUpdated: Date = Date()
    ) {
        precondition(
            (0.0...1.0).contains(confidence),
            "Confidence must be between 0 and 1"
        )
        self.dominantDirection = dominantDirection
        self.missType = missType
        self.isUserDefined = isUserDefined
        self.confidence = confidence
        self.lastUpdated = lastUpdated
    }
}

/// Represents the directional miss bias (simplified for club bias).
/// Matches Android enum: MissBiasDirection (LEFT, STRAIGHT, RIGHT).
enum MissBiasDirection: String, Codable, CaseIterable, Hashable {
    case left
    case straight
    case right
}

File: android/app/src/main/java/com/example/app/domain/navcaddy/models/MissBias.kt
package com.example.app.domain.navcaddy.models

import com.google.gson.annotations.SerializedName

/**
 * Domain model representing a club's miss bias characteristics.
 *
 * Miss bias captures a club's tendency to miss in a particular direction,
 * either user-defined or inferred from shot history. This information
 * feeds into strategy recommendations.
 *
 * Spec reference: player-profile-bag-management.md R4
 * Plan reference: player-profile-bag-management-plan.md Task 2
 *
 * @property dominantDirection Primary direction of misses (LEFT, STRAIGHT, RIGHT)
 * @property missType Optional detailed miss pattern (e.g., SLICE, HOOK)
 * @property isUserDefined True if manually set by user, false if inferred from data
 * @property confidence Confidence in this bias assessment (0.0-1.0)
 * @property lastUpdated Unix timestamp when this bias was last updated
 */
data class MissBias(
    @SerializedName("dominant_direction")
    val dominantDirection: MissBiasDirection,

    @SerializedName("miss_type")
    val missType: MissType? = null,

    @SerializedName("is_user_defined")
    val isUserDefined: Boolean,

    @SerializedName("confidence")
    val confidence: Double,

    @SerializedName("last_updated")
    val lastUpdated: Long = System.currentTimeMillis()
) {
    init {
        require(confidence in 0.0..1.0) { "Confidence must be between 0 and 1" }
    }
}
